pkg/bot/teams_cloud_grpc.go

```diff
@@ -0,0 +1,178 @@
+package bot
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/sirupsen/logrus"
+	"github.com/sourcegraph/conc/pool"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+
+	"github.com/kubeshop/botkube/internal/config/remote"
+	"github.com/kubeshop/botkube/pkg/api/cloudplatform"
+	pb "github.com/kubeshop/botkube/pkg/api/cloudteams"
+)
+
+type grpcCloudTeamsConnector struct {
+	log          logrus.FieldLogger
+	grpcConn     *grpc.ClientConn
+	remoteConfig remote.Config
+
+	agentActivityWorkers *pool.Pool
+	cloudActivityWorkers *pool.Pool
+
+	activityClient pb.CloudTeams_StreamActivityClient
+}
+
+func newGrpcCloudTeamsConnector(log logrus.FieldLogger, url string) (*grpcCloudTeamsConnector, error) {
+	remoteConfig, _ := remote.GetConfig()
+	//if !ok { TODO: enable when config generator will work
+	//	return nil, fmt.Errorf("while getting remote config for %q", config.CloudTeamsCommPlatformIntegration)
+	//}
+	creds := grpc.WithTransportCredentials(insecure.NewCredentials())
+	opts := []grpc.DialOption{
+		creds,
+		grpc.WithStreamInterceptor(cloudplatform.AddStreamingClientCredentials(remoteConfig)),
+		grpc.WithUnaryInterceptor(cloudplatform.AddUnaryClientCredentials(remoteConfig)),
+	}
+
+	conn, err := grpc.Dial(url, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("while creating gRCP connection: %w", err)
+	}
+
+	return &grpcCloudTeamsConnector{
+		log:          log,
+		grpcConn:     conn,
+		remoteConfig: remoteConfig,
+
+		cloudActivityWorkers: pool.New().WithMaxGoroutines(platformMessageWorkersCount),
+		agentActivityWorkers: pool.New().WithMaxGoroutines(platformMessageWorkersCount),
+	}, nil
+}
+
+func (c *grpcCloudTeamsConnector) Shutdown() {
+	c.log.Info("Shutting down Cloud Teams message processor...")
+
+	if c.activityClient != nil {
+		if err := c.activityClient.CloseSend(); err != nil {
+			c.log.WithError(err).Error("Cannot closing gRPC stream activity connection")
+		}
+	}
+
+	if err := c.grpcConn.Close(); err != nil {
+		c.log.WithError(err).Error("Cannot close gRPC connection")
+	}
+
+	c.cloudActivityWorkers.Wait()
+	c.agentActivityWorkers.Wait()
+}
+
+func (c *grpcCloudTeamsConnector) Start(ctx context.Context) error {
+	activityClient, err := pb.NewCloudTeamsClient(c.grpcConn).StreamActivity(ctx)
+	if err != nil {
+		return fmt.Errorf("while initializing gRPC cloud client: %w", err)
+	}
+
+	err = activityClient.Send(&pb.AgentActivity{
+		Req: &pb.AgentActivity_InstanceId{
+			//InstanceId: c.remoteConfig.Identifier, // TODO: enable when config generator will work
+			InstanceId: "foo-123",
```

Comment:
So, `foo-123` will be replaced by a dynamic logic?

Reply:
yes, but it will be done once the config generator will be in place 👍